dragonfly/implementations/PsyllidProvider.py

```diff
@@ -0,0 +1,404 @@
+'''
+'''
+
+from __future__ import absolute_import
+
+# standard imports
+import logging
+import uuid
+import time
+import os
+import json
+from datetime import datetime
+
+# internal imports
+from dripline import core
+from dragonfly.implementations import EthernetProvider
+
+
+__all__ = []
+
+logger = logging.getLogger(__name__)
+
+__all__.append('PsyllidProvider')
+class PsyllidProvider(core.Provider, core.Spime):
+    '''
+    Provider for direct communication with Psyllid 
+    '''
+    def __init__(self,
+                 psyllid_queue='psyllid',
+                 set_condition_list = [],
+                 **kwargs):
+
+	core.Provider.__init__(self, **kwargs)        
+        self.psyllid_queue = psyllid_queue
+        self._set_condition_list = set_condition_list
+        self.status = None
+        self.status_value = None
+               
+        self.channel_dict = {'a': 0, 'b': 1, 'c': 2}
+        self.freq_dict = {'a': None, 'b': None, 'c': None}
+        self.mode = None
+        
+        
+    def _finish_configure(self):
+        try:
+            if self.request_status()==0:
+                self.activate()
+        except:
+            raise core.exceptions.DriplineGenericDAQError('Psyllid instance with this queue is not responding')
+
+        self.mode = 'streaming'
+        try:
+            self.set_default_central_frequencies()
+        except:
```

Comment:
The idea is that if request_status fails to get a response from psyllid there is no psyllid instance running or the queue is wrong. In any case you cannot use it for data acquisition. 
But request status also catches an error so I dont see how this is working right now...
The PsyllidProvider class as opposed to the MultiPsyllidProvider is not intended to be used for the single channel acquisition and needs more changes and testing anyway.
But you re right the exceptions should all be specific.